forem/forem
# Prefer Minimagick for social image generation

## What type of PR is this? (check all applicable)


## Description

This moves the generation of social images from third party htmlimagetocss.com to Minimagick via a worker.

This will reduce costs, allow this functionality to immediately be available on all Forems, and allow for more flexibility in future changes.

This now runs when an article is changed, or when relevant info is changed on a user or organization, and stored as an image in the already-existing column.

The cached third-party generation is still in place for now as a fallback, but this could be removed at a later date — pending some observation on how this all acts in production.

In development it takes about half a second to generate the image — so not something we want to do wastefully, but not expensive if it only changes when key metadata changes.

## Related Tickets & Documents



- Closes https://github.com/forem/forem/issues/19659

## QA Instructions, Screenshots, Recordings

The font is designed to change depending on length, so testing a few titles to check if they fit appropriately would be good.

Here are examples of different titles resulting in slightly different font sizes.

<img width="802" alt="Screenshot 2023-08-02 at 1 44 38 PM" src="https://github.com/forem/forem/assets/3102842/7e6c90a9-8737-4e9b-ac7d-833b29a4f77f">
<img width="802" alt="Screenshot 2023-08-02 at 1 44 22 PM" src="https://github.com/forem/forem/assets/3102842/20936751-4e39-4451-9a4f-68df7bdadef1">


## Added/updated tests?

      have not been included_
